keymanagementrequirements.md

```diff
@@ -0,0 +1,138 @@
+- Publishers SHOULD be able to generate multiple signatures for a single artifact.
+- Publisher admins MUST have a mechanism to revoke signatures to indicate they are no longer trusted. 
+- Trust stores MUST be configurable by the deployer admin.
+- Deployer admins MUST be able to configure trusted entities for individual repositories and targets.
+- Deployers MUST be able to validate signatures on any version of an artifact including whether they have been revoked by the publisher.
+- Signature validation MUST be enforceable in air-gapped environments. 
+
+## Requirements that need further discussion
+### Signing Key Expiry
+In public key cryptography, a pair of private key a.k.a. signing key and public key a.k.a. verification key is generated, and later the public key is certified by a higher key in the hierarchy. The certification of the signing key has a validity period to designate how long it is valid for. This section will discuss tradeoff and recommendations for expiry times and signing key rotation.
+
+### External Timestamp Server
+Time Stamping Authorities (TSAs) defined by RFC3161 provide signed timestamp for a signature in order to prove that the signature was generated during the validity period of a certificate.
+
+#### Scenarios
+With TSA, signature can be considered valid even if the signing cerificate is expired. This technique is widely used by Authenticode with SignTool, NuGet, Adobe Acrobat, and many other industrial products.
+
+In the world of artifacts, including container images, scenarios are
+- Developers sign their artifacts or images with certificates. The certificates MAY have a configurable expiry time.
+- Developers publish their artifacts or images and at a later date stop maintaining them. Content consumers SHOULD be able to verify signatures until they expire and use the artifacts or images
+- Attackers with compromised keys try to sign artifacts or images with timestamps before the key compromise event.
+
+#### Pros and Cons
+
+There are many public TSA servers available on the Internet. The advantages of public timestamp servers are obvious:
+    Public
+    Free
+
+However, those public TSAs also come with disadvantages:
+    Require Internet access for signing
+        It is not really a disadvantage since public TSAs are online services. Devices in the air-gapped environment SHALL access the Internet for timestamp signing services.
+    Out of the control of the signer
+        External dependency
+        Availability is not assured. No SLA on signing.
+            Not all TSAs are available in all regions.
+            Some regions may have high latency.
+        The certificates of TSAs MAY be revoked at any time without notices.
+            The removal of trust of VeriSign broke .NET 5+ NuGet. Thus Microsoft has to disable the package verification with a new release to unblock customers.
```

Comment:
I by no means want to hide how .NET got broken, as it's a great example for why we need fallback, or more reliable TSAs.
I do wonder if we should include these dated references within the specs, vs. discussions. 
Or, is this whole section a discussion?